### PR TITLE
Add R major mode

### PR DESCRIPTION
**Pull request by mihem**
Inital attempt for an R major mode

I added some basic keybinding for an R major mode, some were taken from  @MillesMcBain https://github.com/MilesMcBain/vsconfig/blob/master/settings.json .

`npm run sort-bindings` was run afterwards as recommended.

### CODE REVIEW

**Comment by mihem** [comment]
I also added some basic keybindings for Quarto mode. It's not perfect because Quarto supports multiple languages and some of the keybindings are specific for R (denoted accordingly), but still useful I think.

**Comment by mihem** [comment]
The fold/unfold bindings are redundant. However, it's much hard to type `SPC z  .  o ` than "SPC m f f" keeping in mind it's a common command. Also, it's only in Quarto major mode so probably not relevant for the majority I think.

**Comment by MarcoIeni** [comment]
the diff is huge. Did you change the formatting?

**Comment by mihem** [comment]
You are right, sorry.

Well i ran `npm run sort-bindings` as recommended .. (I may have also once ran  `npm run format` becuase of this commit https://github.com/VSpaceCode/VSpaceCode/pull/304).

Most commits are relatively small, only https://github.com/VSpaceCode/VSpaceCode/pull/332/files is huge, not possible to view in GitHub, but in VSCode you can see that it's mostly the major mode for R shifted to another location.

So sorting is now wrong? What would you recommend?

**Comment by MarcoIeni** [comment]
running `npm run format` formats the code locally for me. Does it work for you? Otherwise I can push the commit.
image

**Comment by mihem** [comment]
ahh this looks much better.

**Comment by mihem** [comment]
Sure. I copied the keybindings from julia, and only removed non-default ones. So the only ones missin were "Actions". And in fact, I could only identify two languages that integrated these "default" keybindings, rust and java. But now they are included in R and Quarto major mode.

File: package.json [hunk 1]
```diff
@@ -6512,6 +6512,707 @@
...
+                                        ]
+                                    },
+                                    {
+                                        "key": "languageId:r",
+                                        "name": "R",
+                                        "type": "bindings",
+                                        "bindings": [
+                                            {
+                                                "key": "d",
+                                                "name": "debugonce",
```
Comment (stevenguh) [inline package.json:6901]:
```suggestion
                                                "name": "Debugonce",
```

File: package.json [hunk 2]
```diff
@@ -6512,6 +6512,707 @@
...
+                                                "type": "command",
+                                                "icon": "debug",
+                                                "command": "r.runCommandWithSelectionOrWord",
+                                                "args": "debugonce($$)"
+                                            },
+                                            {
+                                                "key": "h",
+                                                "name": "help",
```
Comment (stevenguh) [inline package.json:6909]:
```suggestion
                                                "name": "Help",
```

**Review by stevenguh — COMMENTED** [review]
I don't know R and Quarto enough, but overall seems good. Can you also help update documentation like
https://github.com/VSpaceCode/vspacecode.github.io/commit/a57979fa737820c9c33b60562df5d40ff8759355?

Thanks a lot.

**Comment by mihem** [comment]
Sure. 
Done.
Excited to see this in action in VSpaceCode.

**Review by MarcoIeni — APPROVED** [review]
Thank you 🥳

**Comment by mihem** [comment]
@stevenguh @MarcoIeni can you merge the PR or do you need any further stuff by me?

**Comment by stevenguh** [comment]
I’m on the road and I can merge when I am back tomorrow. Thanks for
submitting the PR

**Comment by MarcoIeni** [comment]
Hey mimen, I was waiting for Steven, because I didn't know if he wanted to give another look at the pr 👍

**Pull request merged**